src/dev/GraphiQL.tsx

```diff
@@ -0,0 +1,218 @@
+import { SplashScreen } from "../splash-screen";
+import { AuthenticationScreen } from "../authentication-screen";
+import { Socket as IOSocket } from "socket.io-client";
+import { createEnvironment } from "../relay-environment";
+import { useLogInMutation } from "../chat/log-in-mutation";
+
+const usePcPassword = () =>
+  usePersistedState<string>("pcPassword", {
+    encode: (value) => JSON.stringify(value),
+    decode: (rawValue) => {
+      if (typeof rawValue === "string") {
+        try {
+          const parsedValue = JSON.parse(rawValue);
+          if (typeof parsedValue === "string") {
+            return parsedValue;
+          }
+        } catch (e) {}
+      }
+      return "";
+    },
+  });
+
+type ConnectionMode =
+  | "connected"
+  | "authenticating"
+  | "authenticated"
+  | "connecting"
+  | "disconnected";
+
+export type AuthenticatedRole = "DM" | "Player";
+
+const AuthenticatedAppShellRenderer: React.FC = ({ children }) => {
+  const [isLoggedIn, logIn] = useLogInMutation();
+
+  React.useEffect(() => {
+    logIn();
+  }, [logIn]);
+
+  if (isLoggedIn === false) {
+    return null;
+  }
+
+  return <div>{children}</div>;
+};
+
+const AuthenticatedGraphQLShell: React.FC<{
+  socket: IOSocket;
+  password: string;
+  role: AuthenticatedRole;
+}> = ({ socket, password, role, children }) => {
+  const relayEnvironment = useStaticRef(() => createEnvironment(socket));
+
+  const [connectionMode, setConnectionMode] = React.useState<ConnectionMode>(
+    "connecting"
+  );
+
+  /**
+   * We only use one tab at a time. The others will be disconnected automatically upon opening dungeon-revealer in another tab.
+   * You can still use dungeon-revealer in two tabs by using the incognito mode of the browser.
+   * We do this in order to prevent message/user connect/music sound effect spamming.
+   */
+  React.useEffect(() => {
+    const authenticate = () => {
+      socket.emit("authenticate", {
+        password: password,
+        desiredRole: role === "DM" ? "admin" : "user",
+      });
+    };
+
+    socket.on("connect", () => {
+      setConnectionMode("connected");
+      authenticate();
+    });
+
+    socket.on("authenticated", () => {
+      setConnectionMode("authenticated");
+    });
+
+    socket.on("reconnect", () => {
+      setConnectionMode("authenticating");
+      socket.emit("authenticate", { password: password });
+    });
+
+    socket.on("disconnect", () => {
+      setConnectionMode("disconnected");
+    });
+
+    if (socket.connected) {
+      authenticate();
+    }
+
+    return () => {
+      socket.off("connect");
+      socket.off("reconnecting");
+      socket.off("reconnect");
+      socket.off("reconnect_failed");
+      socket.off("disconnect");
+    };
+  }, [socket, password]);
+
+  if (connectionMode !== "authenticated") {
+    return <SplashScreen text={connectionMode} />;
+  }
+
+  return (
+    <RelayEnvironmentProvider environment={relayEnvironment}>
```

Comment:
We dont need the RelayEnvironment provider in here, so you can remove that stuff